lib/ask-stack-result-view.coffee

```diff
@@ -198,8 +204,30 @@ class AskStackResultView extends ScrollView
           atom.workspace.activatePreviousPane()
           # editor = atom.workspace.activePaneItem
           editor = atom.workspace.getActivePaneItem()
-          editor.insertText(code)
 
+          # Check if an id was passed in to find information about the author and source
+          if id != undefined
+            # Get the attribute author and source information
+            author_src = $("##{id_type}-author-link-#{id}").attr('href');
+            author_name = $("##{id_type}-author-link-#{id}").html();
+            source_src = $("##{id_type}-link-#{id}").attr('href');
+
+            # Transact the following additions so that they are all in one undo instance
+            editor.transact =>
```

Comment:
I found that you can transact a whole section so that they are all contained within one undo - so now both the comments and the inserted code happens as one transactional undo.